packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts

```diff
@@ -76,43 +93,64 @@ function mineOnEvent(contract: Contract) {
 }
 
 beforeAll(async () => {
-  provider = new providers.JsonRpcProvider(rpcEndpoint);
-
-  await Promise.all(
-    [aEngineConfig, bEngineConfig].map(async config => {
-      await DBAdmin.dropDatabase(config);
-      await DBAdmin.createDatabase(config);
-      await DBAdmin.migrateDatabase(config);
-    })
-  );
-
-  a = await Engine.create(aEngineConfig);
-  b = await Engine.create(bEngineConfig);
-
-  const assetHolder = new Contract(
-    ethAssetHolderAddress,
-    ContractArtifacts.EthAssetHolderArtifact.abi,
-    provider
-  );
-  mineOnEvent(assetHolder);
+  try {
+    provider = new providers.JsonRpcProvider(rpcEndpoint);
+
+    await Promise.all(
+      [aEngineConfig, bEngineConfig].map(async config => {
+        await DBAdmin.dropDatabase(config);
+        await DBAdmin.createDatabase(config);
+        await DBAdmin.migrateDatabase(config);
+      })
+    );
+
+    aEngine = await Engine.create(aEngineConfig);
+    bEngine = await Engine.create(bEngineConfig);
+    const aChainService = new ChainService({
+      ...aEngineConfig.chainServiceConfiguration,
+      logger: aEngine.logger,
+    });
+    const bChainService = new ChainService({
+      ...bEngineConfig.chainServiceConfiguration,
+      logger: bEngine.logger,
+    });
+
+    const syncOptions: SyncOptions = {
+      pollInterval: 1_000,
+      timeOutThreshold: 60_000,
+      staleThreshold: 10_000,
+    };
+    a = await Wallet.create(aEngine, aChainService, TestMessageService.create, syncOptions);
+    b = await Wallet.create(bEngine, bChainService, TestMessageService.create, syncOptions);
+
+    TestMessageService.linkMessageServices(a.messageService, b.messageService, aEngine.logger);
+    const assetHolder = new Contract(
+      ethAssetHolderAddress,
+      ContractArtifacts.EthAssetHolderArtifact.abi,
+      provider
+    );
+    mineOnEvent(assetHolder);
+  } catch (error) {
+    console.error('beforeAll failed');
```

Comment:
Should we use a logger instead of `console`?

Reply:
Yup, this is just some debugging code that snuck in.